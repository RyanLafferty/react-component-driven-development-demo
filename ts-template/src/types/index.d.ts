import React from 'react';

// styling modules
declare module '*.(scss|css)' {
  const content: {[className: string]: string};
  export default content;
}

// svgs
declare module "\*.svg" {
  export const ReactComponent: React.SFC<React.SVGProps<SVGSVGElement>>;
  const src: string;  export default src;
}

// jpg/jpeg images
declare module "\*.(jpg|jpeg)" {
  const content: string;
  export default content;
}

// png images
declare module "\*.png" {
  const content: string;
  export default content;
}

// gif images
declare module "\*.gif" {
  const content: string;
  export default content;
}

// json payloads
declare module "\*.json" {
  const content: string;
  export default content;
}
