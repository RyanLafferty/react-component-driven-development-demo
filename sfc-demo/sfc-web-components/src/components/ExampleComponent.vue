<template>
  <div class="container" />
</template>

<script>
export default {
  name: 'ExampleComponent',
};
</script>

<style scoped lang="scss">
  .container {
    background-color: #f036f0;
    border-radius: 50%;
    height: 200px;
    width: 200px;
  };
</style>
