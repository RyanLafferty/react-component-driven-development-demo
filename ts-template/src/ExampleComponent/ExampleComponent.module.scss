.container {
  color: Green;
  font-size: 16px;
};
