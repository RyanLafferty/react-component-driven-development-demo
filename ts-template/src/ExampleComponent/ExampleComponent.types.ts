export interface ExampleComponentProps {
  name: string;
}
