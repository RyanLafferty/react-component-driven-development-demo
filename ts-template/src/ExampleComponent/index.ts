export { default } from './ExampleComponent';
export { ExampleComponentProps } from './ExampleComponent.types';
