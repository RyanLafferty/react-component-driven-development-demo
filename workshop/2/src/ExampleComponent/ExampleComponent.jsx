import React from 'react';
import PropTypes from 'prop-types';

const ExampleComponent = ({ name }) => (
  <div>
    {`Hello ${name}!`}
  </div>
);

ExampleComponent.propTypes = {
  name: PropTypes.string,
};

ExampleComponent.defaultProps = {
  name: 'World',
};

export default ExampleComponent;
