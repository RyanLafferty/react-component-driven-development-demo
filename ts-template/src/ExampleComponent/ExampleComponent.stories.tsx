import React from 'react';
import { withInfo } from '@storybook/addon-info';
import { withKnobs } from '@storybook/addon-knobs';
import { withPerformance } from 'storybook-addon-performance';
import { storiesOf } from '@storybook/react';
import ExampleComponent from './ExampleComponent';
import ExampleComponentREADME from './README.md';

storiesOf('Example Components', module)
  .addDecorator(withKnobs)
  .addDecorator(withInfo)
  .addDecorator(withPerformance)
  .add('ExampleComponent', () => (
    <ExampleComponent name="World" />
  ), { info: ExampleComponentREADME });
