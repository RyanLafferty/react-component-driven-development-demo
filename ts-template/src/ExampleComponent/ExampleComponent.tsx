import React from 'react';
import styles from './ExampleComponent.module.scss';
import { ExampleComponentProps } from './ExampleComponent.types';

const ExampleComponent: React.FC<ExampleComponentProps> = ({ name }) => (
  <div className={styles.container}>
    Hello {name}!
  </div>
);

export default ExampleComponent;
