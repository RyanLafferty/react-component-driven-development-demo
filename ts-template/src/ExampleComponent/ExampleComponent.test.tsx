import React from 'react';
import * as ShallowRenderer from 'react-test-renderer/shallow';
import ExampleComponent from './ExampleComponent';
import { ExampleComponentProps } from './ExampleComponent.types';

describe('ExampleComponent', () => {
  let props: ExampleComponentProps;
  const renderer = ShallowRenderer.createRenderer();

  beforeEach(() => {
    props = {
      name: 'World',
    };
  });

  describe('when required props are provided', () => {
    it('ExampleComponent renders', () => {
      expect(
        renderer.render(<ExampleComponent {...props} />),
      ).toMatchSnapshot();
    });
  });
});
