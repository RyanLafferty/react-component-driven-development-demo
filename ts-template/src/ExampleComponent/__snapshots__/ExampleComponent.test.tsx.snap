// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleComponent when required props are provided ExampleComponent renders 1`] = `
<div
  className="container"
>
  Hello 
  World
  !
</div>
`;
